### ISSUE TITLE
5.1.0.1736. BBS will not launch when no language is defined.

### ISSUE THREAD

**Issue by BloodStoneBBS** [wwivbbs/wwiv#482]
Updated to b1736, Telnet session opens very briefly and then closes. Not even WFC.
Rolled back to 1735. Launches fine.

Edit: Confirmed after testing on Windows 7 and 2008 w32.
Runs fine on Windows 7, will not launch on Server 2008.

**Comment by BloodStoneBBS** [comment]
Works fine on Windows 7. Perhaps something needed on 2008 that is already on Windows 7?

**Comment by wwiv** [comment]
No idea. I don't see anything in the change that is obvious nor don't have
a win2008 machine

**Comment by BloodStoneBBS** [comment]
This could be an issue with Windows Server 2008. 
Running bbs.exe from the command line and I get: "You need the default language installed to run the BBS"

**Comment by wwiv** [comment]
Hmm. Run init and look at the language config..

**Comment by BloodStoneBBS** [comment]
That's what's odd.
In both 1735 and 1736, on Windows 2008, selecting language in INIT shows nothing. I insert and select English but it does not retain the selection on exit.
On the Windows 7, the language is displayed when I select language.

**Comment by wwiv** [comment]
Is this 2008 or 2008r2?

**Comment by BloodStoneBBS** [comment]
2008 std x86

Sent from my Verizon Wireless 4G LTE smartphone

-------- Original message --------

**Comment by wwiv** [comment]
ok, I don't think I can get msvc installed on there to run to debug anything since server 2008 == vista (which is older than win7 and older than the minimum system needed to run msvc 2015 or wwiv).  I'll see what I can do but no promises since this OS may be too old.

**Comment by wwiv** [comment]
https://www.visualstudio.com/en-us/downloads/visual-studio-2015-system-requirements-vs.aspx#1

**Comment by wwiv** [comment]
Doesnt' appear to be Win2008 related since TRI0N saw it too

**Comment by wwiv** [comment]
<TRI0N> Well Inserted one and the default is English, saved it and it's working fine now..

**Comment by TRI0N** [comment]
Using Windows 7 Pro SP1..
Seems that the new build are looking for Language Correctly. I never had Language Set since I installed WWIV it was always blank.

Possible Issues:
1) The check for Language Never Existed
2) The check for Language was Bugged and was never checking.

Possible Solution would be to check if Language is NULL and default to English.
Example: if(language == null || string.IsNullOrWhiteSpace(language)) {default_to_english}

**Comment by BloodStoneBBS** [comment]
Did not have a problem with the Windows 7 install at all but I may have selected the default early on.
Windows Server 2008 however will not retain the change in INIT.
How did you 'insert one'? Specific file needed? Run INIT elevated? (will try this tonight)

UPDATE:  I feel like a dolt.

I ran INIT from an elevated command Inserted english as the default and the language change held and 1736 launched.

I think you fixed something few of knew was broken!

**Comment by TRI0N** [comment]
I inserted one by going into INIT. Opened Language Config to see it was blank (NULL) and then pressed "I" for Insert. The default is English. Pressed ESC to exit and save and all is working fine. However when I initially ran setup setting up my BBS I looked through all the settings and Language Config was always NULL. So whatever junk code you removed looks like it kick started the Language Check. Most likely it will also fix the install issue for Default Langue for new installation. Wish I was at par with your code in the Source but I'm not so I'm just thinking logically.
